scripts/finish-release.ts

```diff
@@ -0,0 +1,221 @@
+  const args = yargs(process.argv.slice(2))
+    .string('release-branch')
+    .default('release-branch', 'main')
+    .string('dev-branch')
+    .default('dev-branch', 'dev')
+    .string('repository')
+    .default('repository', 'aws-amplify/amplify-cli')
+    .boolean('continue')
+    .default('continue', false)
+    .string('merge-branch').argv;
+  return {
+    continue: args.continue,
+    releaseBranch: args['release-branch'],
+    devBranch: args['dev-branch'],
+    mergeBranch: args['merge-branch'],
+    repository: args.repository,
+  };
+}
+
+export async function main() {
+  const { continue: continueArg, releaseBranch, devBranch, repository, mergeBranch: mergeBranchArg } = getArgs();
+  const git = new Git();
+  const rlInterface = rl.createInterface({
+    input: stdin,
+    output: stdout,
+  });
+
+  if (!git.isCleanWorkingTree()) {
+    console.error('Please run the script from a clean working tree');
+    process.exit(2);
+  }
+
+  const remoteOutput = git.remote(true);
+  const upstreamName = extractUpstreamNameFromRemotes(remoteOutput, repository);
+  if (!upstreamName) {
+    console.error('could not find remote name for the aws-amplify/amplify-cli repository');
+    process.exit(1);
+  }
+
+  console.info(`Fetching ${upstreamName}/${releaseBranch}`);
+  git.fetch(upstreamName, releaseBranch);
+
+  const releaseSha = git.getShortSha(`${upstreamName}/${releaseBranch}`);
+  const mergeBranch = mergeBranchArg ?? 'dev-main-merge-' + releaseSha;
+  const doesMergeBranchExist = git.isExistingBranch(mergeBranch);
+
+  if (!continueArg && doesMergeBranchExist) {
+    console.error(
+      `The merge branch \`${mergeBranch}\` already exists. Please delete the merge branch and try again, or run the script with \`--continue\``,
+    );
+    process.exit(1);
+  }
+  const canContinue = continueArg || (await shouldContinue(rlInterface, `This will create the merge branch \`${mergeBranch}\`.`));
+  if (!canContinue) {
+    console.info('Exiting...');
+    process.exit(0);
+  }
+
+  git.checkout(devBranch);
+  git.pull();
+  console.info('Creating merge branch ' + mergeBranch);
+  try {
+    const shouldAttemptCreate = !doesMergeBranchExist;
+    git.checkout(mergeBranch, shouldAttemptCreate);
+  } catch (e) {
+    console.error('Failed to create merge branch, exiting...');
+    process.exit(1);
+  }
+
+  console.info(`Merging ${upstreamName}/${releaseBranch} into ${mergeBranch}`);
+  git.merge(`${upstreamName}/${releaseBranch}`, { message: 'chore: merge release commit from main to dev' });
+
+  if (!git.isCleanWorkingTree()) {
+    console.info('Resolve merge conflicts and commit merge, then run `yarn run finish-release --continue`');
+    process.exit(0);
+  }
+
+  git.push(upstreamName, mergeBranch);
+
+  console.info(`
+You can compare changes between the branches here: ${getCompareLink(
+    repository,
+    devBranch,
+    mergeBranch,
+  )}. To finish the release, open a pull request between ${mergeBranch} into ${devBranch}
```

Comment:
we should try https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request